edx_oauth_client/pipeline.py

```diff
@@ -1,139 +1,164 @@
-AUTH_ENTRY_LOGIN_API = 'login_api'
-AUTH_ENTRY_REGISTER_API = 'register_api'
-
 
 @partial.partial
 def ensure_user_information(
-        strategy, auth_entry, backend=None, user=None, social=None,
-        allow_inactive_user=False, *args, **kwargs):
+        strategy, auth_entry, backend=None, user=None, social=None, allow_inactive_user=False, *args, **kwargs
+):
     """
-    Ensure that we have the necessary information about a user (either an
-    existing account or registration data) to proceed with the pipeline.
+    Ensure that we have the necessary information about a user to proceed with the pipeline.
+
+    Either an existing account or registration data.
     """
 
-    response = {}
     data = {}
-
     try:
         if 'data' in kwargs['response']:
             user_data = kwargs['response']['data'][0]
         else:
             user_data = kwargs['response']
+        log.info('Get user data')
 
-        log.info('Get user data: %s', str(user_data))
         data['access_token'] = kwargs['response']['access_token']
 
+        country = user_data.get('country', 'UA')
+
+        if not country:
+            log.info('No country in response.')
+
         for key, value in backend.setting('USER_DATA_KEY_VALUES').items():
             data[key] = user_data.get(value)
 
-        if 'username' not in data or not data['username']:
-            data['username'] = data.get('name', slugify(data['email']))
+        if not data['email']:
+            data['email'] = strategy.session_get('email', "")
 
-        if any((data['first_name'], data['last_name'])):
-            data['name'] = u'{} {}'.format(data['first_name'], data['last_name']).strip()
-        else:
-            data['name'] = user_data.get('preferred_username')
+        data['username'] = data['email']
 
-        if not all((data['username'], data['email'])):
+        if kwargs.get('is_new') and not all((data['username'], data['email'])):
             raise AuthEntryError(
                 backend,
-                'One of the required parameters (username or email) is not received with the user data.'
+                "One of the required parameters (username or email) is not received with the user data."
             )
     except AuthEntryError as e:
         log.exception(e)
         raise
     except Exception as e:
         log.exception(e)
-        raise AuthEntryError(backend, 'Cannot receive user\'s data')
-
-    def dispatch_to_register():
-        """Force user creation on login or register"""
+        raise AuthEntryError(backend, "Cannot receive user's data")
 
-        request = strategy.request
-        data.update({
-            'terms_of_service': 'True',
-            'honor_code': 'True',
-            'password': make_random_password()
-        })
-
-        if request.session.get('ExternalAuthMap'):
-            del request.session['ExternalAuthMap']
+    if not user:
```

Comment:
Do we need to pass this parameters to 'data' and then to the `AccountCreationForm` if user is not exist?

Reply:
Resolved